tests/filesTest.js

```diff
@@ -202,345 +337,362 @@ fdescribe('Main: Currently testing files management,', function () {
-      expect(status2).toEqual(true)
-      return oc.files.fileInfo(newFile, ['{http://owncloud.org/ns}favorite'])
-    }).then(fileInfo => {
-      expect(fileInfo.getProperty('{http://owncloud.org/ns}favorite')).toEqual('1')
-      return oc.files.favorite(newFile, false)
-    }).then(status2 => {
-      expect(status2).toEqual(true)
-      return oc.files.fileInfo(newFile, ['{http://owncloud.org/ns}favorite'])
-    }).then(fileInfo => {
-      expect(fileInfo.getProperty('{http://owncloud.org/ns}favorite')).toEqual('0')
-      return oc.files.delete(newFile)
-    }).then(status2 => {
-      expect(status2).toEqual(true)
-      done()
-    }).catch(error => {
-      fail(error)
-      done()
+    it('checking method: getFileUrl', function () {
+      const url = oc.files.getFileUrl('/foo/bar')
+      expect(url).toBe(owncloudURL + 'remote.php/webdav/foo/bar')
     })
-  })
 
-  it('checking method: favorite filter', function (done) {
-    const newFile = config.testFolder + '/' + 'file.txt'
-
-    oc.files.putFileContents(newFile, testContent).then(status => {
-      expect(typeof status).toBe('object')
-      return oc.files.favorite(newFile)
-    }).then(status2 => {
-      expect(status2).toEqual(true)
-      return oc.files.getFavoriteFiles(['{http://owncloud.org/ns}favorite'])
-    }).then(files => {
-      expect(files.length).toEqual(1)
-      expect(files[0].getProperty('{http://owncloud.org/ns}favorite')).toEqual('1')
-      return oc.files.delete(newFile)
-    }).then(status2 => {
-      expect(status2).toEqual(true)
-      done()
-    }).catch(error => {
-      expect(error).toBe(null)
-      done()
+    it('checking method: getFileUrlV2', function () {
+      const url = oc.files.getFileUrlV2('/foo/bar')
+      expect(url).toBe(owncloudURL + 'remote.php/dav/files/admin/foo/bar')
     })
-  })
+    it('checking method : mkdir for an existing parent path', async function (done) {
+      const newFolder = testFolder + '/' + 'new folder'
+      await provider.addInteraction(deleteResource(encodeURI(newFolder)))
 
-  fit('checking method : mkdir for an existing parent path', function (done) {
-    const newFolder = config.testFolder + '/' + 'new folder/'
-
-    oc.files.mkdir(newFolder).then(status => {
-      expect(status).toBe(true)
-      return oc.files.list(newFolder, 0)
-    }).then(folder => {
-      folder = folder[0]
-      expect(folder.isDir()).toBe(true)
-      expect(folder.getName()).toEqual('new folder')
-      return oc.files.delete(newFolder)
-    }).then(status2 => {
-      expect(status2).toEqual(true)
-      done()
-    }).catch(error => {
-      expect(error).toBe(null)
-      done()
+      oc.files.mkdir(newFolder).then(status => {
+        expect(status).toBe(true)
+        return oc.files.list(newFolder, 0)
+      }).then(folder => {
+        folder = folder[0]
+        expect(folder.isDir()).toBe(true)
+        expect(folder.getName()).toEqual('new folder')
+        return oc.files.delete(newFolder)
```

Comment:
no need to delete the folder after the test, this test is about mkdir and there is no real server running where the folder is created. so this "delete" test should be moved to a separate test, or delete it completely if there are other explicit delete tests (can be done in a separate PR)